app/src/main/java/org/example/ChessBoard.java

```diff
@@ -0,0 +1,32 @@
+package org.example;
+
+import org.example.ChessPieces.ChessPiece;
+import org.example.ChessPieces.ChessPieceGenerator;
+
+public class ChessBoard {
+    private ChessPiece mActivePiece;
+    private ChessPiece[][] board;
+
+    ChessBoard(String[] boardLayout) {
+
+        ChessPieceGenerator generator = new ChessPieceGenerator();
+
+        board = new ChessPiece[8][8];
+
+        for (int i = 0; i < board.length; i++) {
+            for (int j = 0; j < board[i].length; j++) {
+                String currPiece = boardLayout[i].substring(j, j + 1);
```

Comment:
It's better to require the boardLayout to be 8x8 as opposed to accept 10x10 and work around it with `j + 1`.

It should be the caller's responsibility to pass 8x8 even if the caller stores it internally as 10x10.

In other words, the GUI layout shouldn't influence your chess board layout.

Reply:
I think there's some confusion here. `j + 1` is there to extract one character from `boardLayout` as a String, not to work around 8x8 vs 10x10 grids.

Currently the board can be any size, but I don't know how to 'require' `boardLayout` to be 8x8, except by just knowing this and passing the correct length variable. 

Also as of now there's no logical relationship between the GUI board and the logical board. For now I'll just keep the `grid` as 10x10, and the `boardLayout` as 8x8.